### DISCUSSION TITLE
use less buttons and more gestures

### DISCUSSION THREAD

**Discussion by tharvik** [Abrynos/ShoppingList#117] [Ideas]
I was thinking about reworking some of the interactions with the items. here goes a few ideas, tell me what you think of it

* remove the bin button to use a swipe-to-delete gesture
  * there seems to be an easy way to achieve that using an [`ItemTouchHelper`](https://developer.android.com/reference/androidx/recyclerview/widget/ItemTouchHelper)
* allow to edit an item (change name & amount) when long pressing on it
  * pretty much reuse the add activity to also be an edit one
* remove the plus & minus button from the main activity (as available in the edit activity)
  * I don't find to be changing the number of items regularly, I usually set it up once before making my shopping
* short press on an item crosses it
  * after some reflexion, simply removing it, or setting it to zero can be a bit bothering when misclicking. it should be easily reversable by short pressing again
    * I'm still not sure if we should offer a way to clear the list of crossout items and how to present it nicely; maybe a floating button on the bottom left? short to remove crossout, long to remove all?
  * that would be a bit more involving code-wise, as we need to keep track of the crossout state in addition to the name & amount (to allow to revert click)

**Comment by Abrynos** [comment]
> remove the bin button to use a swipe-to-delete gesture

I like this idea and might look into it myself when I'm back home. However, this behaviour should only be active if the "system navigation" setting is "gesture navigation". If the user has selected "2-button navigation" or "3-button navigation", we should keep our current setup. [potential detection algorithm](https://gist.github.com/Thorsten1976/07d61b3f697364e5f1c08ae076641d58). For going back from the add/options activity we already support gesture navigation (in case gesture navigation setting is active in the android system).

Since, we're not requiring a minimum android API version fo 30 but instead 25 this whole thing might prove a bit trickier than you expect it to be.

> allow to edit an item (change name & amount) when long pressing on it

This has already been suggested in #75. Feel free to send a merge request. We already "kind of" support this. E.g. if you press the "send" button and send a link to someone else living in your household. When they click on the link, the add-activity is opened with the contents pre-filled and editable. However, we'd need to adapt the titlebar, which explicitly states "Add Item". When sending a merge-request, please make sure to unify the logic... I have an idea on how to implement it nicely, but I'd have to check.

> remove the plus & minus button from the main activity

To this I have to say "out of scope". I myself regularly use this feature for things usually bought in bigger quantities (or when on sale). If I go low on stocks, I'll add it to the list and every time I use up another instance of my remaining stocks, I'll increase the quantity. That way, when going shopping there is an indicator of urgency for my purchase. Plus I know at least one other person who handles their list this way.

> short press on an item crosses it

As items with an amount of `0` are already greyed out (and can be ordered last). The suggested advantage of "crossing out" from other feature requests does not exist any more.

**Reply by Abrynos** [reply]
Update: I've started editing items on a new local branch.
